tests/Feature/CreateDatabaseTest.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace Tests\Feature;
+
+use App\Actions\CreateDatabase;
+use App\Shell\Shell;
+use Illuminate\Support\Facades\Config;
+use Symfony\Component\Process\ExecutableFinder;
+use Tests\Feature\Fakes\FakeProcess;
+use Tests\TestCase;
+
+class CreateDatabaseTest extends TestCase
+{
+    /** @test */
+    public function it_creates_a_mysql_database()
+    {
+        $this->fakeLamboConsole();
+
+        Config::set('lambo.store.create_database', true);
+        Config::set('lambo.store.database_username', 'user');
+        Config::set('lambo.store.database_password', 'password');
+        Config::set('lambo.store.database_name', 'database_name');
+
+        $this->mock(Shell::class, function ($shell) {
+            $shell->shouldReceive('execInProject')
+                ->with('mysql --user=user --password=password -e "CREATE DATABASE IF NOT EXISTS database_name";')
+                ->once()
+                ->andReturn(FakeProcess::success());
+        });
+        $this->assertEquals('Created a new database ' . Config::get('lambo.store.database_name'), app(CreateDatabase::class)());
+    }
+
+    /** @test */
+    public function it_checks_if_mysql_is_installed()
+    {
+        $this->fakeLamboConsole();
+
+        Config::set('lambo.store.create_database', true);
+        Config::set('lambo.store.database_username', 'user');
+        Config::set('lambo.store.database_password', 'password');
+        Config::set('lambo.store.database_name', 'database_name');
+
+        $this->mock(ExecutableFinder::class, function($executableFinder){
+            $executableFinder->shouldReceive('find')
+                ->with('mysql')
+                ->once()
+                ->andReturn(null);
+        });
+
+        $this->assertEquals('MySql does not seem to be installed. Skipping new database creation.', app(CreateDatabase::class)());
```

Comment:
Could we check something a bit sturdier, like whether the call is ever made to the exec, or something else?

Reply:
This is a weird one. I had it like that before. I was doing:

```php
$this->mock(Shell::class, function ($shell) {
    $shell->shouldNotReceive('execInProject')
          ->with('mysql --user=user --password=password -e "CREATE DATABASE IF NOT EXISTS database_name";');
});
```

The test was passing but then I noticed that if I changed it to `$shell->shouldReceive...` it still passed. Thats when I changed the test to assert against the returned value. At least its something?

I did look at the Mockery docs and google for people having similar issues but I didn't find anything.